### ISSUE TITLE
Pulse: slack messaging not working - NullpointerException

### ISSUE THREAD

**Issue by joerg-etherapists** [metabase/metabase#7986]
We cannot use pulses via slack. The slack integration seems to be fine (no error while saving the API Token of slack). 
It was working in the past - unfortunately we  don't know exactly when it was broken.
We tested it in the pulse section with the button "Send to Slack now" and it tells "Sending failed". 
See the stracktrace below.

- Browser: Chrome Version 67.0.3396.99 (64-Bit), Firefox  60.0.2 (64-Bit)
- OS:  Ubuntu 17.10
- Databases:  MySQL 5.5.60
- Metabase version:  0.29.0
- Metabase hosting environment: (Linux/Ubuntu 17.10, Java: openjdk version "1.8.0_171" )
- Metabase internal database: MySQL 5.5.60

Stracktrace in the logfile service.log:

```
2018-07-06 11:48:55,555 [qtp1682623231-508] ERRORmetabase.middleware - POST /api/pulse/test 500 (5 s) (41 DB calls).
{:message nil,
 :type java.lang.NullPointerException,
 :stacktrace
 ["pulse.render$render_to_png.invokeStatic(render.clj:334)"
  "pulse.render$render_to_png.invoke(render.clj:317)"
  "pulse.render$fn__23557$render_html_to_png__23562$fn__23566.invoke(render.clj:352)"
  "pulse.render$fn__23557$render_html_to_png__23562.invoke(render.clj:344)"
  "pulse.render$render_pulse_card_to_png.invokeStatic(render.clj:836)"
  "pulse.render$render_pulse_card_to_png.invoke(render.clj:833)"
  "pulse$create_slack_attachment_data$iter__32336__32340$fn__32341$fn__32342$fn__32345.invoke(pulse.clj:70)"
  "pulse$create_and_upload_slack_attachments_BANG_$iter__32359__32363$fn__32364$fn__32365.invoke(pulse.clj:81)"
  "pulse$create_and_upload_slack_attachments_BANG_$iter__32359__32363$fn__32364.invoke(pulse.clj:80)"
  "pulse$create_and_upload_slack_attachments_BANG_.invokeStatic(pulse.clj:79)"
  "pulse$create_and_upload_slack_attachments_BANG_.invoke(pulse.clj:76)"
  "pulse$fn__32447.invokeStatic(pulse.clj:202)"
  "pulse$fn__32447.invoke(pulse.clj:200)"
  "pulse$send_notifications_BANG_.invokeStatic(pulse.clj:215)"
  "pulse$send_notifications_BANG_.invoke(pulse.clj:213)"
  "pulse$send_pulse_BANG_.invokeStatic(pulse.clj:241)"
  "pulse$send_pulse_BANG_.doInvoke(pulse.clj:232)"
  "api.pulse$fn__38933$fn__38936.invoke(pulse.clj:156)"
  "api.common.internal$do_with_caught_api_exceptions.invokeStatic(internal.clj:254)"
  "api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj:249)"
  "api.pulse$fn__38933.invokeStatic(pulse.clj:148)"
  "api.pulse$fn__38933.invoke(pulse.clj:148)"
  "middleware$enforce_authentication$fn__38506.invoke(middleware.clj:118)"
  "api.routes$fn__49410.invokeStatic(routes.clj:82)"
  "api.routes$fn__49410.invoke(routes.clj:82)"
  "routes$fn__49469$fn__49470.doInvoke(routes.clj:108)"
  "routes$fn__49469.invokeStatic(routes.clj:103)"
  "routes$fn__49469.invoke(routes.clj:103)"
  "middleware$log_api_call$fn__38605$fn__38607.invoke(middleware.clj:349)"
  "middleware$log_api_call$fn__38605.invoke(middleware.clj:348)"
  "middleware$add_security_headers$fn__38555.invoke(middleware.clj:251)"
  "core$wrap_streamed_json_response$fn__54821.invoke(core.clj:67)"
  "middleware$bind_current_user$fn__38510.invoke(middleware.clj:139)"
  "middleware$maybe_set_site_url$fn__38559.invoke(middleware.clj:275)"]}
```

**Comment by salsakran** [comment]
can you try sending it manually? if that doesn't work, can you try adding and removing cards until it does? It's bombing out on a render, and it would be very useful to know what kind of card it's bombing out on

**Comment by domparry** [comment]
I have the same issue here, just trying to show a standard table.

Running Metabase v 0.31.2, but had this problem on 0.31.1 too (It's a new installation).

Pulses do work for email though. metabot is present in slack, and responds to requests to list questions. I get the null pointer exception requesting the show from slack, or manually triggering the pulse from metabase.

See stack trace below:

```
java.lang.NullPointerException
	at sun.awt.FontConfiguration.getVersion(FontConfiguration.java:1264)
	at sun.awt.FontConfiguration.readFontConfigFile(FontConfiguration.java:219)
	at sun.awt.FontConfiguration.init(FontConfiguration.java:107)
	at sun.awt.X11FontManager.createFontConfiguration(X11FontManager.java:774)
	at sun.font.SunFontManager$2.run(SunFontManager.java:431)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.font.SunFontManager.<init>(SunFontManager.java:376)
	at sun.awt.FcFontManager.<init>(FcFontManager.java:35)
	at sun.awt.X11FontManager.<init>(X11FontManager.java:57)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at sun.font.FontManagerFactory$1.run(FontManagerFactory.java:83)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.font.FontManagerFactory.getInstance(FontManagerFactory.java:74)
	at java.awt.Font.getFont2D(Font.java:491)
	at java.awt.Font.getFamily(Font.java:1220)
	at java.awt.Font.getFamily_NoClientCode(Font.java:1194)
	at java.awt.Font.getFamily(Font.java:1186)
	at org.fit.cssbox.layout.VisualContext.update(VisualContext.java:249)
	at org.fit.cssbox.layout.VisualContext.updateForGraphics(VisualContext.java:348)
	at org.fit.cssbox.layout.ElementBox.loadBasicStyle(ElementBox.java:1293)
	at org.fit.cssbox.layout.ElementBox.setStyle(ElementBox.java:315)
	at org.fit.cssbox.layout.InlineBox.setStyle(InlineBox.java:92)
	at org.fit.cssbox.layout.BoxFactory.createElementInstance(BoxFactory.java:825)
	at org.fit.cssbox.layout.BoxFactory.createBox(BoxFactory.java:744)
	at org.fit.cssbox.layout.BoxFactory.createElementBox(BoxFactory.java:447)
	at org.fit.cssbox.layout.BoxFactory.createSubtree(BoxFactory.java:263)
	at org.fit.cssbox.layout.BoxFactory.createViewportTree(BoxFactory.java:169)
	at org.fit.cssbox.layout.BrowserCanvas.createLayout(BrowserCanvas.java:169)
	at org.fit.cssbox.layout.BrowserCanvas.createLayout(BrowserCanvas.java:142)
	at metabase.pulse.render$render_to_png.invokeStatic(render.clj:325)
	at metabase.pulse.render$render_to_png.invoke(render.clj:301)
	at metabase.pulse.render$fn__25987$render_html_to_png__25992$fn__25996.invoke(render.clj:336)
	at metabase.pulse.render$fn__25987$render_html_to_png__25992.invoke(render.clj:328)
	at metabase.pulse.render$render_pulse_card_to_png.invokeStatic(render.clj:839)
	at metabase.pulse.render$render_pulse_card_to_png.invoke(render.clj:836)
	at metabase.pulse$create_slack_attachment_data$iter__40385__40389$fn__40390$fn__40391$fn__40394.invoke(pulse.clj:69)
	at metabase.pulse$create_and_upload_slack_attachments_BANG_$iter__40408__40412$fn__40413$fn__40414.invoke(pulse.clj:80)
	at metabase.pulse$create_and_upload_slack_attachments_BANG_$iter__40408__40412$fn__40413.invoke(pulse.clj:79)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:528)
	at clojure.core$seq__5124.invokeStatic(core.clj:137)
	at clojure.core$dorun.invokeStatic(core.clj:3125)
	at clojure.core$doall.invokeStatic(core.clj:3140)
	at clojure.core$doall.invoke(core.clj:3140)
	at metabase.pulse$create_and_upload_slack_attachments_BANG_.invokeStatic(pulse.clj:78)
	at metabase.pulse$create_and_upload_slack_attachments_BANG_.invoke(pulse.clj:75)
	at metabase.pulse$fn__40496.invokeStatic(pulse.clj:201)
	at metabase.pulse$fn__40496.invoke(pulse.clj:199)
	at clojure.lang.MultiFn.invoke(MultiFn.java:229)
	at metabase.pulse$send_notifications_BANG_$fn__40507.invoke(pulse.clj:217)
	at metabase.pulse$send_notifications_BANG_.invokeStatic(pulse.clj:216)
	at metabase.pulse$send_notifications_BANG_.invoke(pulse.clj:212)
	at metabase.pulse$send_pulse_BANG_.invokeStatic(pulse.clj:245)
	at metabase.pulse$send_pulse_BANG_.doInvoke(pulse.clj:236)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at metabase.api.pulse$fn__55135.invokeStatic(pulse.clj:191)
	at metabase.api.pulse$fn__55135.invoke(pulse.clj:181)
	at compojure.core$make_route$fn__16249.invoke(core.clj:135)
	at compojure.core$wrap_route_middleware$fn__16242.invoke(core.clj:122)
	at compojure.core$wrap_route_info$fn__16246.invoke(core.clj:126)
	at compojure.core$if_route$fn__16198.invoke(core.clj:45)
	at compojure.core$if_method$fn__16188.invoke(core.clj:27)
	at compojure.core$routing$fn__16256.invoke(core.clj:151)
	at clojure.core$some.invokeStatic(core.clj:2693)
	at clojure.core$some.invoke(core.clj:2684)
	at compojure.core$routing.invokeStatic(core.clj:151)
	at compojure.core$routing.doInvoke(core.clj:148)
	at clojure.lang.RestFn.applyTo(RestFn.java:139)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$apply.invoke(core.clj:652)
	at compojure.core$routes$fn__16260.invoke(core.clj:156)
	at metabase.middleware$enforce_authentication$fn__56174.invoke(middleware.clj:113)
	at compojure.core$routing$fn__16256.invoke(core.clj:151)
	at clojure.core$some.invokeStatic(core.clj:2693)
	at clojure.core$some.invoke(core.clj:2684)
	at compojure.core$routing.invokeStatic(core.clj:151)
	at compojure.core$routing.doInvoke(core.clj:148)
	at clojure.lang.RestFn.invoke(RestFn.java:423)
	at metabase.api.routes$fn__56350.invokeStatic(routes.clj:77)
	at metabase.api.routes$fn__56350.invoke(routes.clj:77)
	at compojure.core$if_context$fn__16280.invoke(core.clj:220)
	at compojure.core$routing$fn__16256.invoke(core.clj:151)
	at clojure.core$some.invokeStatic(core.clj:2693)
	at clojure.core$some.invoke(core.clj:2684)
	at compojure.core$routing.invokeStatic(core.clj:151)
	at compojure.core$routing.doInvoke(core.clj:148)
	at clojure.lang.RestFn.applyTo(RestFn.java:139)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$apply.invoke(core.clj:652)
	at compojure.core$routes$fn__16260.invoke(core.clj:156)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:657)
	at clojure.core$apply.invoke(core.clj:652)
	at metabase.routes$fn__56409$fn__56410.doInvoke(routes.clj:108)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at compojure.core$routing$fn__16256.invoke(core.clj:151)
	at clojure.core$some.invokeStatic(core.clj:2693)
	at clojure.core$some.invoke(core.clj:2684)
	at compojure.core$routing.invokeStatic(core.clj:151)
	at compojure.core$routing.doInvoke(core.clj:148)
	at clojure.lang.RestFn.invoke(RestFn.java:423)
	at metabase.routes$fn__56409.invokeStatic(routes.clj:103)
	at metabase.routes$fn__56409.invoke(routes.clj:103)
	at compojure.core$if_context$fn__16280.invoke(core.clj:220)
	at compojure.core$routing$fn__16256.invoke(core.clj:151)
	at clojure.core$some.invokeStatic(core.clj:2693)
	at clojure.core$some.invoke(core.clj:2684)
	at compojure.core$routing.invokeStatic(core.clj:151)
	at compojure.core$routing.doInvoke(core.clj:148)
	at clojure.lang.RestFn.applyTo(RestFn.java:139)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$apply.invoke(core.clj:652)
	at compojure.core$routes$fn__16260.invoke(core.clj:156)
	at clojure.lang.Var.invoke(Var.java:381)
	at metabase.middleware$catch_api_exceptions$fn__56309.invoke(middleware.clj:436)
	at metabase.middleware$log_api_call$fn__56287$fn__56289.invoke(middleware.clj:364)
	at toucan.db$_do_with_call_counting.invokeStatic(db.clj:203)
	at toucan.db$_do_with_call_counting.invoke(db.clj:196)
	at metabase.middleware$log_api_call$fn__56287.invoke(middleware.clj:363)
	at metabase.middleware$add_security_headers$fn__56229.invoke(middleware.clj:252)
	at ring.middleware.json$wrap_json_body$fn__62462.invoke(json.clj:44)
	at metabase.core$wrap_streamed_json_response$fn__62673.invoke(core.clj:67)
	at ring.middleware.keyword_params$wrap_keyword_params$fn__62519.invoke(keyword_params.clj:36)
	at ring.middleware.params$wrap_params$fn__62567.invoke(params.clj:67)
	at metabase.middleware$bind_current_user$fn__56179.invoke(middleware.clj:137)
	at clojure.core$comp$fn__5529.invoke(core.clj:2561)
	at clojure.core$comp$fn__5529.invoke(core.clj:2561)
	at clojure.core$comp$fn__5529.invoke(core.clj:2561)
	at metabase.middleware$maybe_set_site_url$fn__56239.invoke(middleware.clj:290)
	at puppetlabs.i18n.core$locale_negotiator$fn__124.invoke(core.clj:357)
	at ring.middleware.cookies$wrap_cookies$fn__62401.invoke(cookies.clj:175)
	at ring.middleware.session$wrap_session$fn__62658.invoke(session.clj:108)
	at metabase.middleware$add_content_type$fn__56232.invoke(middleware.clj:262)
	at ring.middleware.gzip$wrap_gzip$fn__62432.invoke(gzip.clj:65)
	at ring.adapter.jetty$proxy_handler$fn__62260.invoke(jetty.clj:25)
	at ring.adapter.jetty.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:531)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:352)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:281)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:762)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:680)
	at java.lang.Thread.run(Thread.java:745)```

**Comment by mazameli** [comment]
Changing priority to P1 to match #10088, which may be a duplicate, though the error messages look different to me.

**Comment by salsakran** [comment]
at the very very least, we need better error handling for pulse send failures.

**Comment by maxzheng** [comment]
Thanks @flyingmachine for the hint. Issue is resolved after upgrading from OpenJDK8 to 11 for me.

**Comment by pan-christensen** [comment]
Changing from headless to normal version of java 8 solved the issue.

**Comment by camsaul** [comment]
This was addressed by #12543
